src/transaction.js

```diff
@@ -251,8 +252,8 @@ Transaction.prototype.clone = function () {
  * hashType, and then hashes the result.
  * This hash can then be used to sign the provided transaction input.
  */
-Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashType) {
-  typeforce(types.tuple(types.UInt32, types.Buffer, /* types.UInt8 */ types.Number), arguments)
+Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashType, inAmount) {
```

Comment:
Maybe it'd be better to take the approach used for SegWit,  a different function not `hashForSignature`,  avoids positional parameters